airbyte-webapp/src/components/JobItem/components/Logs.tsx

```diff
@@ -57,4 +61,38 @@ const Logs: React.FC<LogsProps> = ({ logsArray }) => {
   );
 };
 
+/**
+ * Matching the log's line number by time makes the following assumptions:
+ * 1. The log's lines are already ordered by time
+ * 2. The timestamps used are in the same timezone
+ * 3. The error is closer to the end of the log file than the beginning
+ * 4. Lines are matched by the start of a line with "YYYY-MM-DD HH:mm:ss" format, like "2022-05-17 23:00:19 DEBUG I am a log message"
+ *
+ * TODO: it appears that LazyLog only highlights the first 2 lines in the array.  Can this be fixed?
+ */
+const getMatchingLineNumbers = (matchTimestamp: number | undefined, lines: string[] | undefined) => {
+  if (!matchTimestamp || !lines || lines.length === 0) {
+    return [];
+  }
+
+  const matchingLineNumbers: number[] = [];
+  const matcher = /^[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1]) (2[0-3]|[01][0-9]):[0-5][0-9]:[0-5][0-9]/;
+
+  let lineCounter = lines.length - 1;
+  while (lineCounter >= 0) {
+    const timeString = lines[lineCounter].match(matcher);
+    if (timeString) {
+      const datetime = Date.parse(`${timeString[0].replace(" ", "T")}Z`);
+      if (datetime - 1000 <= matchTimestamp && datetime + 1000 >= matchTimestamp) {
+        matchingLineNumbers.push(lineCounter + 1);
+      } else if (datetime - 2001 <= matchTimestamp) {
```

Comment:
note: we'll be highlighting the log lines that are within +-1second of the trace message.